bug: e2e helper buildKukeRunPathArgs puts --no-daemon at root, broken by #567's flag retirement

## Context

PR #567 (commit 7612bd5, \"refactor: retire --no-daemon from workload commands per #222\") removed `--no-daemon` from `rootCmd.PersistentFlags()`. PR #564 (commit d16040f, \"test(e2e): make --no-daemon explicit in buildKukeRunPathArgs\") changed `e2e/e2e_test.go:173` to prepend `--no-daemon` at root level:

```go
func buildKukeRunPathArgs(runPath string) []string {
    return []string{\"--no-daemon\", \"--run-path\", runPath}
}
```

The combination is incompatible: every e2e invocation now shells out as `./kuke --no-daemon --run-path <X> <subcmd> ...`, and cobra mis-parses the unknown root-level `--no-daemon` flag — the next non-flag token is then interpreted as the subcommand. Observed failure (any e2e test):

```
=== CONT  TestKuke_NoRealms
    e2e_kuke_realm_test.go:132: running /root/kukeon/kuke [--no-daemon --run-path /tmp/TestKuke_NoRealms3365628405/001 get realm --output json] failed: exit status 1
        output:
        Error: unknown command \"/tmp/TestKuke_NoRealms3365628405/001\" for \"kuke\"
        Run 'kuke --help' for usage.
--- FAIL: TestKuke_NoRealms (0.01s)
```

The bug is invisible to CI's \`make test\` because the \`test\` target excludes \`./e2e\` (\`go test \$(shell go list ./... | grep -v /e2e)\`). It is exactly the failure mode CLAUDE.md §PR creation's \"literal-flag-string grep\" rule warns about — removing a CLI flag without scanning paths excluded from the CI test target (e2e/, scripts/, docs/) — and the grep was apparently skipped on #567. Reproduces on \`origin/main@7612bd5\` against any e2e test.

## Proposal

Drop \`--no-daemon\` from \`buildKukeRunPathArgs\` (e2e/e2e_test.go:173). \`--run-path\` already implies \`--no-daemon\` via \`applyRunPathImpliesNoDaemon\` in cmd/kuke/kuke.go (the helper's own comment block calls this out as \"belt-and-suspenders\"), so the prefix only needs the one flag. Update the comment to drop the \"two flags travel together\" framing accordingly.

Alternative: re-introduce \`--no-daemon\` as a root-persistent flag. Rejected — defeats the entire point of #222 / #567, which deliberately demoted the flag to per-command local registration.

## Acceptance criteria

- [ ] \`buildKukeRunPathArgs\` returns \`[\"--run-path\", runPath]\` only.
- [ ] \`TestKuke_NoRealms\` (and any other helper-dependent e2e test) passes on a host with docker + containerd up, against the local kukeon-local:e2e image, with no other source changes.
- [ ] Comment block on the helper is updated to reflect the new single-flag prefix; the \"belt-and-suspenders\" framing is dropped.
- [ ] No other call sites use the literal \`\"--no-daemon\"\` string preceded by a root-cmd invocation (grep \`e2e/ scripts/ docs/\` to confirm).

## Out of scope

- Re-adding \`--no-daemon\` as a root-persistent flag (rejected above).
- The Makefile e2e target's other env plumbing (covered separately under #557).

## References

- Surfaced verifying #557 (\"nested-aware e2e socket\"). Discovered when running \`TestKuke_Init_VerifyState\` after applying the nested socket fix — the run never reached the socket-collision path because the cobra parse failed earlier.
- PR #564 (d16040f) introduced the helper layout.
- PR #567 (7612bd5) removed \`--no-daemon\` from \`rootCmd.PersistentFlags()\`.
- cmd/kuke/kuke.go:367 \`applyRunPathImpliesNoDaemon\` — the auto-promotion that makes the explicit \`--no-daemon\` redundant.

--- a/e2e/e2e_test.go
+++ b/e2e/e2e_test.go
@@ -157,21 +157,23 @@ func deepSocketPathFits(runPath string) bool {
 	return len(runPath)+deepSocketSuffixLen <= consts.KukeonMaxSocketPath
 }
 
-// buildKukeRunPathArgs returns the canonical `--no-daemon --run-path <X>`
-// prefix every e2e invocation must carry. The two flags travel together
-// because the e2e suite has no shared kukeond running on its --run-path:
-// without --no-daemon, kuke would dial the host's daemon socket (whichever
-// /opt/kukeon path it was started against) and read/write someone else's
-// state. Per-test --run-path + in-process controller is the only mode that
-// keeps the suite hermetic on hosts where containerd is up but kukeond is
-// not (the common e2e harness shape).
+// buildKukeRunPathArgs returns the canonical `--run-path <X>` prefix every
+// e2e invocation must carry. The e2e suite has no shared kukeond running on
+// its --run-path; without the in-process promotion that --run-path triggers,
+// kuke would dial the host's daemon socket (whichever /opt/kukeon path it
+// was started against) and read/write someone else's state. Per-test
+// --run-path + in-process controller is the only mode that keeps the suite
+// hermetic on hosts where containerd is up but kukeond is not (the common
+// e2e harness shape).
 //
-// Note: --run-path alone already implies --no-daemon at the CLI layer via
-// applyRunPathImpliesNoDaemon (cmd/kuke/kuke.go), so this prefix is
-// belt-and-suspenders. Spelling both out keeps test source self-
-// documenting and severs the dependency on that promotion.
+// The in-process promotion comes from applyRunPathImpliesNoDaemon
+// (cmd/kuke/kuke.go), which auto-sets --no-daemon=true whenever --run-path
+// is explicit. The flag is no longer accepted at root (#222 / #567 retired
+// it from workload commands and demoted it to per-command local
+// registration), so spelling it out here would now mis-parse as an unknown
+// root flag and shift the subcommand token.
 func buildKukeRunPathArgs(runPath string) []string {
-	return []string{"--no-daemon", "--run-path", runPath}
+	return []string{"--run-path", runPath}
 }
 
 // loadKukeondImageIntoContainerd stages the local kukeond image into the

